enterprise/reporting/error_reporting.go

```diff
@@ -317,82 +338,97 @@ func (edr *ErrorDetailReporter) mainLoop(ctx context.Context, c types.SyncerConf
 	edr.errorDetailReportsQueryTime = edr.stats.NewTaggedStat("error_detail_reports_query_time", stats.TimerType, tags)
 	edr.edReportingRequestLatency = edr.stats.NewTaggedStat("error_detail_reporting_request_latency", stats.TimerType, tags)
 
-	// In infinite loop
-	// Get Reports
-	// Aggregate
-	// Send in a separate go-routine
-	// Delete in a separate go-routine
-	for {
-		if ctx.Err() != nil {
-			edr.log.Infof("stopping mainLoop for syncer %s : %s", c.Label, ctx.Err())
-			return
-		}
-		requestChan := make(chan struct{}, edr.maxConcurrentRequests.Load())
-		loopStart := time.Now()
-		currentMs := time.Now().UTC().Unix() / 60
+	var lastReportedAtTime atomic.Time
+	lastReportedAtTime.Store(time.Now())
```

Comment:
We need to update `lastReportedAtTime` every time we flush reports to reporting. We can use the output from [this](https://github.com/rudderlabs/rudder-server/pull/5028/files?diff=split&w=1#diff-6067339d2725ff3da3cb6c33d9b28a5becfb9278bd1e5ff14a364bece14584b7R470) query to update the `lastReportedAtTime`

Reply:
This is done. Can you check once ?